janitor/utils.py

```diff
@@ -456,160 +442,90 @@ def _check_instance(entry: Dict):
-                or isinstance(value.columns, pd.MultiIndex)
+            if (isinstance(value.index, pd.MultiIndex)) or (
+                isinstance(value.columns, pd.MultiIndex)
             ):
-                # add key to dataframe columns
-                value = value.add_prefix(f"{key}_")
-                dfs.append(value)
-            else:
                 raise TypeError(
                     "`expand_grid` does not work with pd.MultiIndex"
                 )
         # process lists
-        if isinstance(value, list):
+        if isinstance(value, (list, tuple)):
             if not value:
                 raise ValueError("passed Sequence cannot be empty")
-            if np.array(value).ndim == 1:
-                checklist = (type(None), str, int, float, bool)
-                instance_check_type = (
-                    isinstance(internal, checklist) for internal in value
-                )
-                if all(instance_check_type):
-                    dicts.update({key: value})
-                else:
-                    raise ValueError("values in iterable must be scalar")
-            elif np.array(value).ndim == 2:
-                value = pd.DataFrame(value).add_prefix(f"{key}_")
-                dfs.append(value)
-            else:
-                raise ValueError("Sequence's dimension should be 1d or 2d")
 
-    return dfs, dicts
+    return entry
 
 
-def _grid_computation_dict(dicts: Dict) -> pd.DataFrame:
-    """
-    Function used within the expand_grid function,
-    to compute dataframe from values that are not dataframes/arrays/series.
-    These values are collected into a dictionary,
-    and processed with numpy meshgrid.
-
-    Numpy's meshgrid is faster than itertools' product -
-    the speed difference shows up,
-    as the size of the input dictionary increases,
-    and is also fast when converting to a dataframe.
-    """
-    # if there is only name value pair in the dictionary
-    if len(dicts) == 1:
-        final = pd.DataFrame(dicts)
-    # if there are more than one name value pair
-    else:
-        # extract value from each key:value pair
-        # in the dicts dictionary
-        extracted_data = [value for key, value in dicts.items()]
-        # create the cartesian product of the extracted data
-        res = np.meshgrid(*extracted_data)
-
-        # get sorter array from the first entry in the res list
-        # this way, we can safely sort all
-        # the arrays by the first array
-        sorter = np.argsort(res[0].ravel())
-        # flatten each array and sort with the sorter
-        res = [entry.ravel()[sorter] for entry in res]
-        # pair each array with the appropriate dictionary key
-        res = dict(zip(dicts, res))
-
-        # create dataframe
-        final = pd.DataFrame(res)
-    return final
-
-
-def _grid_computation_list(frames: List) -> pd.DataFrame:
+def _grid_computation(entry: Dict) -> pd.DataFrame:
```

Comment:
I think we might still need a descriptive docstring for this function. The original was quite helpful for deciphering what was going on. It'll also help in case anybody wants to import the function and use it directly outside of pandas/pyjanitor.